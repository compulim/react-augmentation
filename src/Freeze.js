import PropTypes from 'prop-types';
import React     from 'react';

export default class Freeze extends React.PureComponent {
  constructor(props, context) {
    super(props, context);

    this.state = {
      children: props.frozen && props.children
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.frozen !== this.props.frozen) {
      this.setState(() => ({
        children: nextProps.frozen && nextProps.children
      }));
    }
  }

  render() {
    const { frozen, ...otherProps } = this.props;

    return (
      <div { ...otherProps }>
        {
          this.props.frozen ?
            this.state.children
          :
            this.props.children
        }
      </div>
    );
  }
}

Freeze.propTypes = {
  frozen: PropTypes.bool
};
