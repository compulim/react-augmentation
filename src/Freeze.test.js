import React    from 'react';
import renderer from 'react-test-renderer';

import Freeze from './Freeze';

it('passthru when not frozen', () => {
  const component = renderer.create(
    <Freeze>
      <p>Hello, World!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();

  component.update(
    <Freeze className="passed">
      <p>Aloha!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('frozen and thaw correctly', () => {
  const component = renderer.create(
    <Freeze>
      <p>Hello, World!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();

  component.update(
    <Freeze frozen={ true }>
      <p>Hello, World!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();

  component.update(
    <Freeze frozen={ true }>
      <p>Aloha!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();

  component.update(
    <Freeze frozen={ false }>
      <p>Aloha!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('initial freeze correctly', () => {
  const component = renderer.create(
    <Freeze frozen={ true }>
      <p>Hello, World!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();

  component.update(
    <Freeze frozen={ true }>
      <p>Aloha!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('freeze multiple children correctly', () => {
  const component = renderer.create(
    <Freeze frozen={ true }>
      <p>How are you?</p>
      <p>I am fine.</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();

  component.update(
    <Freeze frozen={ true }>
      <p>Hello, World!</p>
      <p>Aloha!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();

  component.update(
    <Freeze frozen={ false }>
      <p>Hello, World!</p>
      <p>Aloha!</p>
    </Freeze>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('should freeze with reordered key correctly', () => {
  const component = renderer.create(
    <div>
      <Freeze key={ 0 } frozen={ true }>
        <p>How are you?</p>
      </Freeze>
      <Freeze key={ 1 } frozen={ true }>
        <p>I am fine</p>
      </Freeze>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();

  component.update(
    <div>
      <Freeze key={ 1 } frozen={ true }>
        <p>Hello, World!</p>
      </Freeze>
      <Freeze key={ 0 } frozen={ true }>
      </Freeze>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
