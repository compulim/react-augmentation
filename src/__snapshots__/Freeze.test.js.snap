// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`freeze multiple children correctly 1`] = `
<div>
  <p>
    How are you?
  </p>
  <p>
    I am fine.
  </p>
</div>
`;

exports[`freeze multiple children correctly 2`] = `
<div>
  <p>
    How are you?
  </p>
  <p>
    I am fine.
  </p>
</div>
`;

exports[`freeze multiple children correctly 3`] = `
<div>
  <p>
    Hello, World!
  </p>
  <p>
    Aloha!
  </p>
</div>
`;

exports[`frozen and thaw correctly 1`] = `
<div>
  <p>
    Hello, World!
  </p>
</div>
`;

exports[`frozen and thaw correctly 2`] = `
<div>
  <p>
    Hello, World!
  </p>
</div>
`;

exports[`frozen and thaw correctly 3`] = `
<div>
  <p>
    Hello, World!
  </p>
</div>
`;

exports[`frozen and thaw correctly 4`] = `
<div>
  <p>
    Aloha!
  </p>
</div>
`;

exports[`initial freeze correctly 1`] = `
<div>
  <p>
    Hello, World!
  </p>
</div>
`;

exports[`initial freeze correctly 2`] = `
<div>
  <p>
    Hello, World!
  </p>
</div>
`;

exports[`passthru when not frozen 1`] = `
<div>
  <p>
    Hello, World!
  </p>
</div>
`;

exports[`passthru when not frozen 2`] = `
<div
  className="passed"
>
  <p>
    Aloha!
  </p>
</div>
`;

exports[`should freeze with reordered key correctly 1`] = `
<div>
  <div>
    <p>
      How are you?
    </p>
  </div>
  <div>
    <p>
      I am fine
    </p>
  </div>
</div>
`;

exports[`should freeze with reordered key correctly 2`] = `
<div>
  <div>
    <p>
      I am fine
    </p>
  </div>
  <div>
    <p>
      How are you?
    </p>
  </div>
</div>
`;
